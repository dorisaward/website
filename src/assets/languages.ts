import { I18n } from 'aws-amplify/utils'

export const setLanguage = I18n.setLanguage

export const getText = (key: Dictionary) => I18n.get(key)

export enum languages {
    en = 'ENGLISH',
    th = 'THAI',
  }

type Dictionary = 'sorry'

type Dictionaries = Record<languages, Record<Dictionary, string>>

export const dictionary: Dictionaries = {
    [languages.en]: {
        'sorry': 'Sorry'
    },
    [languages.th]: {
        'sorry': 'ขอโทษ'
    },
}

export const setUpLanguages = () => {
    I18n.putVocabularies(dictionary)
    setLanguage(languages.en)
}
