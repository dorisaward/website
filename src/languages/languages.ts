export enum languages {
    en = 'ENGLISH',
    th = 'THAI',
}
