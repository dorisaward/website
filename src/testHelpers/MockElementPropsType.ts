import { ReactNode } from 'react'

export type MockElementPropsType<P = unknown> = P & { children?: ReactNode | undefined }
