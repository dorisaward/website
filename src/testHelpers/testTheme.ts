import { DefaultTheme } from 'styled-components'

export const testTheme: DefaultTheme = {
    backgroundColour: 'backgroundColour',
    fontColour: 'fontColour',
    padding: 'padding',
    secondaryColour: 'secondaryColour',
    shadowColour: 'shadowColour',
    switchOffColour: 'switchOffColour',
    switchOnColour: 'switchOnColour',
    tertiaryColour: 'tertiaryColour'
}