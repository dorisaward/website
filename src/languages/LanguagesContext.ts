import { createContext } from 'react'
import { languages } from './languages.ts'
import { LangContext } from './types/LanguagesContext.ts'

export const LanguagesContext = createContext<LangContext>({
    language: languages.en,
    getText: () => '',
    handleLangPress: () => {},
})