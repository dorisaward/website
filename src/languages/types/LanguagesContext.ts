import { languages } from '../languages.ts'
import { Dictionary } from './Dictionary.ts'

export type LangContext = {
    language: languages
    getText: (key: Dictionary) => string
    handleLangPress: () => void
}