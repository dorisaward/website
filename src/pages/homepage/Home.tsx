import styled, { useTheme } from "styled-components"
import { Daw } from "../../assets/Daw"

const Text = styled.p`
    color: ${({ theme }) => theme.fontColour}
`

export const Home = () => {
    const theme = useTheme()
    return (
        <>
            <Daw width={'50%'} stroke={theme.fontColour} />
            <Text>
                Under construction
            </Text>
        </>
    )
}