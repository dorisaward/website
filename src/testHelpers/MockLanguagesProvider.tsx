import { MockElementPropsType } from './MockElementPropsType.ts'
import { LanguagesContext } from '../languages/LanguagesContext.ts'
import { LangContext } from '../languages/types/LanguagesContext.ts'

export const MockLanguagesProvider = ({ children, value }: MockElementPropsType<{ value: LangContext }>) => (
    <LanguagesContext value={value}>
        {children}
    </LanguagesContext>
)
