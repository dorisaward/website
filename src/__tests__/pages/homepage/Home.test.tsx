import { LangContext } from '../../../languages/types/LanguagesContext.ts'
import { languages } from '../../../languages/languages.ts'
import { render } from '@testing-library/react'
import { MockLanguagesProvider } from '../../../testHelpers/MockLanguagesProvider.tsx'
import { Home } from '../../../pages/homepage/Home.tsx'
import { testTheme } from '../../../testHelpers/testTheme.ts'
import { ThemeProvider } from 'styled-components'
import { MockElementPropsType } from '../../../testHelpers/MockElementPropsType.ts'

jest.mock('../../../assets/Daw.tsx', () => ({
    Daw: ({ children, ...restProps }: MockElementPropsType) => <div data-testid='dawTestId' {...restProps}>{children}</div>
}))

describe('Home', () => {
    it('renders', () => {
        // Given
        const testText = 'testText'
        const value: LangContext = {
            getText: jest.fn(() => testText),
            handleLangPress: jest.fn(),
            language: languages.en
        }

        // When
        const { getByText, getByTestId, baseElement } = render(
            <ThemeProvider theme={testTheme}>
                <MockLanguagesProvider value={value}>
                    <Home/>
                </MockLanguagesProvider>
            </ThemeProvider>
        )

        // Then
        expect(getByText(testText)).toBeTruthy()
        expect(getByTestId('dawTestId')).toBeTruthy()
        expect(baseElement).toMatchSnapshot()
    })
})