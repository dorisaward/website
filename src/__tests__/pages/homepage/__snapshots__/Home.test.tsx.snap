// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders 1`] = `
<body>
  <div>
    <div
      class="sc-blHHSb jweioB"
    >
      <div
        data-testid="dawTestId"
        stroke="fontColour"
        width="50%"
      />
      <p
        class="sc-gtLWhw krnjwF"
      >
        testText
      </p>
    </div>
  </div>
</body>
`;
